spec.html

```diff
@@ -26777,88 +26777,83 @@ <h1>Overview of Date Objects and Definitions of Abstract Operations</h1>
-        <p>A Number can exactly represent all integers from -9,007,199,254,740,992 to 9,007,199,254,740,992 (<emu-xref href="#sec-number.min_safe_integer"></emu-xref> and <emu-xref href="#sec-number.max_safe_integer"></emu-xref>). A time value supports a slightly smaller range of exactly -100,000,000 days to 100,000,000 days measured relative to midnight at the beginning of 01 January, 1970 UTC. This yields an exact supported time value range of -8,640,000,000,000,000 to 8,640,000,000,000,000 milliseconds relative to midnight at the beginning of 01 January, 1970 UTC.</p>
+        <p>An ECMAScript <dfn>time value</dfn> is a Number, either a finite integer representing an instant in time to millisecond precision or *NaN* representing no specific instant.
+        <p>Time is measured in ECMAScript as milliseconds since midnight at the beginning of 01 January, 1970 UTC. Time in ECMAScript does not observe leap seconds; they are ignored. Time calculations assume each and every day contains exactly <emu-eqn id="eqn-msPerDay" aoid="msPerDay">msPerDay = 60 &times; 60 &times; 24 &times; 1000 = 86,400,000</emu-eqn> milliseconds, to align with the POSIX specification of each and every day containing exactly 86,400 seconds.</p>
+        <p>A Number can exactly represent all integers from -9,007,199,254,740,992 to 9,007,199,254,740,992 (<emu-xref href="#sec-number.min_safe_integer"></emu-xref> and <emu-xref href="#sec-number.max_safe_integer"></emu-xref>). A time value used for the [[DateValue]] internal slot of a Date instance supports a slightly smaller range of -8,640,000,000,000,000 to 8,640,000,000,000,000 milliseconds (exactly -100,000,000 days to 100,000,000 days relative to midnight at the beginning of 01 January, 1970 UTC).</p>
         <p>The exact moment of midnight at the beginning of 01 January, 1970 UTC is represented by the time value *+0*.</p>
         <emu-note>
           <p>The 400 year cycle of the Gregorian calendar contains 97 leap years. This yields an average of 365.2425 days per year, or an average of 31,556,952,000 milliseconds per year under the Gregorian calendar. ECMAScript applies a proleptic Gregorian calendar for all time computations.</p>
           <p>As specified by this section, the maximum year range a Number can represent exactly with millisecond precision is approximately -285,426 to 285,426 years relative to midnight at the beginning of 01 January, 1970 UTC.</p>
-          <p>As specified by this section, the maximum year range a time value can represent is approximately -273,790 to 273,790 years relative to midnight at the beginning of 01 January, 1970 UTC.</p>
+          <p>As specified by this section, the maximum year range a [[DateValue]] can represent is approximately -273,790 to 273,790 years relative to midnight at the beginning of 01 January, 1970 UTC.</p>
```

Comment:
Right, and it's actually a critically important aspect here that some time values fall outside the valid range of the [[DateValue]] internal slot of Date instances.

For a simple example, consider LocalTime, which returns a time value with UTC date and time of day equal to the local date and time of day at the instant in time specified by its time value argument. This is equivalent to returning the the sum of its argument and the local UTC offset at that instant, which can exceed the maximum valid [[DateValue]] if the argument is already close to it (e.g., if the local UTC offset at the maximum [[DateValue]] of +275760-09-13T00:00Z is +10:00, then LocalTime(8_640_000_000_000_000) returns 8_640_000_036_000_000). And similar effects apply with negative UTC offsets near the minimum [[DateValue]] boundary.

Reply:
I think that would confuse rather than illuminate. There really is just one concept, but the full range of safe integers is only available for internal operations to avoid rounding errors where the spec calls for UTC offset conversion/DST calculation/etc. The bounds are enforced by TimeClip, which is introduced as the last operation in this overview section. Perhaps the range aspects of "Time Values and Time Range" should be pulled into an explicit subsection and TimeClip should be moved there, making it the first operation rather than the last?

Reply:
I like that suggestion, @gibson042.

Reply:
Done.